src/utils.ts

```diff
@@ -0,0 +1,74 @@
+import * as fs from "fs";
+import * as core from "@actions/core";
+import * as ini from "ini";
+import * as path from "path";
+import * as os from "os";
+
+export function checkStorageDriver(filePaths: string[]): void {
+    const filePath = checkStorageFile(filePaths);
+    if (filePath) {
+        core.info("Reading storage file to find out the storage driver used");
+        const fileContent = ini.parse(fs.readFileSync(filePath, "utf-8"));
+        const storageDriver = fileContent.storage.driver;
+        core.info(`Storage driver is set to "${storageDriver}"`);
+
+        // Check for storage-driver, if overlay then and
+        // add mount program as "fuse-overlayfs" in ~/.config/containers/storage.conf
+        if (storageDriver === "overlay") {
+            // Backup file if ~/.config/containers/storage.conf exists
+            if (filePath === filePaths[0]) {
```

Comment:
Since the logic for [0] and [1] (home dir vs /etc/) is fairly different, maybe it should be two different function calls or functions because this is hard to follow

Reply:
Remove the array and have two strings for home/etc locations instead and then refactor these two functions accordingly

it would be necessary for https://github.com/redhat-actions/buildah-build/pull/51/files#r609678313 too